scripts/abcidump/cmd/capture.go

```diff
@@ -0,0 +1,142 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
+	"github.com/spf13/cobra"
+
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/scripts/abcidump/parser"
+)
+
+const (
+	flagPort      = "port"
+	flagInterface = "interface"
+	flagPromisc   = "promisc"
+
+	flagRequestType  = "request-type"
+	flagResponseType = "response-type"
+)
+
+// CaptureCmd captures network traffic and dumps it as JSON message
+type CaptureCmd struct {
+	cmd       *cobra.Command
+	Input     io.Reader
+	Promisc   bool
+	Port      uint16
+	Interface string
+
+	RequestType  string
+	ResponseType string
+}
+
+// Command returns cobra command
+func (captureCmd *CaptureCmd) Command() *cobra.Command {
+	if captureCmd.cmd != nil {
+		return captureCmd.cmd
+	}
+
+	captureCmd.cmd = &cobra.Command{
+		Use:   "capture",
+		Short: "capture TCP traffic, parse as ABCI protocol and display it in JSON format",
+
+		RunE: captureCmd.RunE,
+		PostRun: func(cmd *cobra.Command, args []string) {
+			if captureCmd.Input != nil {
+				if closer, ok := captureCmd.Input.(io.Closer); ok {
+					closer.Close()
+				}
+			}
+		},
+	}
+
+	captureCmd.cmd.Flags().StringVarP(&captureCmd.Interface, flagInterface, "i", "eth0", "interface to use")
+	captureCmd.cmd.Flags().Uint16VarP(&captureCmd.Port, flagPort, "p", 26658, "Port number to use")
+	captureCmd.cmd.Flags().StringVar(&captureCmd.RequestType, flagRequestType,
+		"tendermint.abci.Request", "Protobuf type of the request")
+	captureCmd.cmd.Flags().StringVar(&captureCmd.ResponseType, flagResponseType,
+		"tendermint.abci.Response", "Protobuf type of the response")
+	captureCmd.cmd.Flags().BoolVar(&captureCmd.Promisc, flagPromisc, false, "Enable promiscuous mode")
+
+	return captureCmd.cmd
+}
+
+// RunE executes traffic capture
+func (captureCmd *CaptureCmd) RunE(cmd *cobra.Command, args []string) error {
+	logger.Debug("Starting packet capture", "port", captureCmd.Port)
+
+	return captureCmd.capture()
+}
+
+func (captureCmd *CaptureCmd) capture() error {
+	handle, err := pcap.OpenLive(captureCmd.Interface, 65535, captureCmd.Promisc, time.Second)
+	if err != nil {
+		return err
+	}
+	defer handle.Close()
+
+	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
+	for packet := range packetSource.Packets() {
+		if err := captureCmd.processPacket(packet); err != nil {
+			logger.Error("cannot parse packet", "error", err)
+		}
+	}
+
+	return nil
+}
+
+func (captureCmd *CaptureCmd) processPacket(packet gopacket.Packet) error {
+	var err error
```

Comment:
it looks like you forgot to remove this variable declaration.
just below need to modify on this
```
err := p.Parse(msgType)
```